packages/formik/src/types.tsx

```diff
@@ -53,118 +55,281 @@ export interface FormikState<Values> {
+}
+
+export type FormikReducerState<Values> = FormikInitialState<Values> &
+  FormikCurrentState<Values>;
+
+export type FormikMessage<
+  Values,
+  State extends FormikReducerState<Values> = FormikReducerState<Values>
+> =
+  | { type: 'SUBMIT_ATTEMPT' }
+  | { type: 'SUBMIT_FAILURE' }
+  | { type: 'SUBMIT_SUCCESS' }
+  | { type: 'SET_ISVALIDATING'; payload: boolean }
+  | { type: 'SET_ISSUBMITTING'; payload: boolean }
+  | { type: 'SET_VALUES'; payload: Values }
+  | { type: 'RESET_VALUES'; payload: Values }
+  | { type: 'SET_FIELD_VALUE'; payload: { field: string; value?: any } }
+  | { type: 'SET_FIELD_TOUCHED'; payload: { field: string; value?: boolean } }
+  | { type: 'SET_FIELD_ERROR'; payload: { field: string; value?: string } }
+  | { type: 'SET_TOUCHED'; payload: FormikTouched<Values> }
+  | { type: 'RESET_TOUCHED'; payload: FormikTouched<Values> }
+  | { type: 'SET_ERRORS'; payload: FormikErrors<Values> }
+  | { type: 'RESET_ERRORS'; payload: FormikErrors<Values> }
+  | { type: 'SET_STATUS'; payload: any }
+  | { type: 'RESET_STATUS'; payload: any }
+  | {
+      type: 'SET_FORMIK_STATE';
+      payload: (s: State) => State;
+    }
+  | {
+      type: 'RESET_FORM';
+      payload: State;
+    };
+
 /**
- * Formik computed properties. These are read-only.
+ * Formik computed state. These are read-only and
+ * result from updates to FormikState but do not live there.
  */
-export interface FormikComputedProps<Values> {
-  /** True if any input has been touched. False otherwise. */
+export interface FormikComputedState {
+  /**
+   * True if `!isEqual(initialValues, state.values)`
+   */
   readonly dirty: boolean;
-  /** True if state.errors is empty */
+  /**
+   * True if one of:
+   * `dirty && state.errors is empty` or
+   * `!dirty && isInitialValid`
+   */
   readonly isValid: boolean;
-  /** The initial values of the form */
-  readonly initialValues: Values;
-  /** The initial errors of the form */
-  readonly initialErrors: FormikErrors<Values>;
-  /** The initial visited fields of the form */
-  readonly initialTouched: FormikTouched<Values>;
-  /** The initial status of the form */
-  readonly initialStatus?: any;
 }
 
+/**
+ * @deprecated use FormikComputedState
+ */
+export type FormikComputedProps = FormikComputedState;
+
+export type FormikState<Values> = FormikReducerState<Values> & FormikComputedState;
+
+export type GetStateFn<
+  Values
+> = () => FormikState<Values>;
+export type UnregisterFieldFn = (name: string) => void;
+export type RegisterFieldFn = (name: string, { validate }: any) => void;
+
 /**
  * Formik state helpers
  */
+
+export type SetStatusFn = (status: any) => void;
+
+export type SetErrorsFn<Values extends FormikValues> = (
+  errors: FormikErrors<Values>
+) => void;
+
+export type SetSubmittingFn = (isSubmitting: boolean) => void;
+
+export type SetTouchedFn<Values extends FormikValues> = (
+  touched: import('./types').FormikTouched<Values>,
```

Comment:
why is this `import()` here? is this importing itself? that works???

Reply:
I think vs code autocompleted this... And somehow it works I guess?! Lol